src/json.ts

```diff
@@ -266,3 +272,166 @@ export function getJsonRpcIdValidator(options?: JsonRpcValidatorOptions) {
+   *
+   * This function assumes the encoding of the JSON is done in UTF-8.
+   *
+   * @param value - Potential JSON serializable value.
+   * @param skipSizing - Skip JSON size calculation (default: false).
+   * @returns Tuple [isValid, plainTextSizeInBytes] containing a boolean that signals whether
+   * the value was serializable and a number of bytes that it will use when serialized to JSON.
+   */
+  function getJsonSerializableInfo(
+    value: unknown,
+    skipSizing: boolean,
+  ): [isValid: boolean, plainTextSizeInBytes: number] {
+    if (value === undefined) {
+      // Return zero for undefined, since these are omitted from JSON serialization
+      return [true, 0];
+    } else if (value === null) {
+      // Return already specified constant size for null (special object)
+      return [true, skipSizing ? 0 : JsonSize.Null];
+    }
+
+    // Check and calculate sizes for basic types
+    // eslint-disable-next-line default-case
+    switch (typeof value) {
+      case 'function':
+        return [false, 0];
+      case 'string':
+        return [
+          true,
+          skipSizing ? 0 : calculateStringSize(value) + JsonSize.Quote * 2,
+        ];
+      case 'boolean':
+        if (skipSizing) {
+          return [true, 0];
+        }
+        return [true, value ? JsonSize.True : JsonSize.False];
+      case 'number':
+        return [true, skipSizing ? 0 : calculateNumberSize(value)];
+    }
+
+    // Handle specific complex objects that can be serialized properly
+    try {
+      if (value instanceof Date) {
+        if (skipSizing) {
+          return [true, 0];
+        }
+        const jsonSerializedDate = value.toJSON();
+        return [
+          true,
+          // Note: Invalid dates will serialize to null
+          jsonSerializedDate === null
+            ? JsonSize.Null
+            : calculateStringSize(jsonSerializedDate) + JsonSize.Quote * 2,
+        ];
+      } else if (value instanceof Boolean) {
+        if (skipSizing) {
+          return [true, 0];
+        }
+        // eslint-disable-next-line eqeqeq
+        return [true, value == true ? JsonSize.True : JsonSize.False];
+      } else if (value instanceof Number) {
+        if (skipSizing) {
+          return [true, 0];
+        }
+        return [true, value.toString().length];
+      } else if (value instanceof String) {
+        if (skipSizing) {
+          return [true, 0];
+        }
+        return [
+          true,
+          calculateStringSize(value.toString()) + JsonSize.Quote * 2,
+        ];
+      }
+    } catch (_) {
+      return [false, 0];
+    }
+
+    // If object is not plain and cannot be serialized properly,
+    // stop here and return false for serialization
+    if (!isPlainObject(value) && !Array.isArray(value)) {
+      return [false, 0];
+    }
+
+    // Handle circular objects
+    if (seenObjects.indexOf(value) !== -1) {
+      return [false, 0];
+    }
+    seenObjects.push(value);
```

Comment:
Does this work if I have the same object in multiple places but not circular? This should be valid:

```
const date = new Date();
{ a: date, b: date }
```

Reply:
Yea, I just realized it would not work that way... It worked for Date object, but not for multiple plain... So, I've improved the solution and added one tricky test which will test the scenario you described.